var/spack/repos/builtin/packages/exago/package.py

```diff
@@ -101,15 +114,16 @@ class Exago(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("hiop@0.3.99:", when="@0.99:+hiop")
     depends_on("hiop@0.5.1:", when="@1.1.0:+hiop")
     depends_on("hiop@0.5.3:", when="@1.3.0:+hiop")
+    depends_on("hiop@0.7.0:", when="@1.5.0:+hiop")
 
     depends_on("hiop~mpi", when="+hiop~mpi")
     depends_on("hiop+mpi", when="+hiop+mpi")
 
     depends_on("petsc@3.13:3.14", when="@:1.2.99")
     depends_on("petsc@3.16.0:3.16", when="@1.3.0:")
-    depends_on("petsc~mpi", when="~mpi")
+    depends_on("petsc@3.18.0:3.18", when="@1.5.0:")
```

Comment:
I think the the above petsc dependencies are conflicting (with each other).. Perhaps you intend to specify:

```
    depends_on("petsc@3.16.0:3.16", when="@1.3.0:1.4")
    depends_on("petsc@3.16.0:3.18", when="@1.5.0:")
```

Reply:
BTW: I'm currently using:
```
+    depends_on("petsc@3.16.0:3.16", when="@1.3.0:1.4")
+    depends_on("petsc@3.18.0:3.18", when="@1.5.0:")
```